src/engine.rs

```diff
@@ -80,12 +81,11 @@ pub fn run(function_path: PathBuf, input_path: PathBuf) -> Result<FunctionRunRes
     let logs =
         std::str::from_utf8(&logs).map_err(|e| anyhow!("Couldn't print Function Logs: {}", e))?;
 
-    let output = output_stream
+    let raw_output = output_stream
         .try_into_inner()
         .expect("Output stream reference still exists")
         .into_inner();
-    let output: serde_json::Value = serde_json::from_slice(output.as_slice())
-        .map_err(|e| anyhow!("Couldn't decode Function Output: {}", e))?;
+    let output = serde_json::from_slice(raw_output.as_slice()).unwrap_or_default();
```

Comment:
If we fail to parse the output as JSON, we might want to show the user the raw string value instead of null so they can more easily debug what's happening. I'm thinking of a case where the function has a bug in it that causes it to output invalid JSON.

Reply:
+1. Also I wonder if it makes sense to use `serde_json::Value::Null` as the default here because that is also a possible output from the function. What if we had an enum like this:
```rust
enum FunctionOutput<'a> {
    Json(serde_json::Value),
    Other(&'a [u8]), // not sure what the best value is here
}
```
and then for the `Other` variant we could output it but maybe using red on the header with a note that it was invalid JSON

Reply:
The very last thing a Function does is returning its result, if the Function fails before returning, it will no return anything. "Returning" here is printing to stdout. However, if the value returned is not valid JSON, then an other error will occur. I forgot about this edge case and I'll update this PR with a fix. I really like the FunctionOutput enum idea, I'll play with that also.